### PR TITLE
feat(source-map-debug): Add improved source map debug endpoint (blue thunder edition)

### LINKED ISSUE #55664: Source Maps Debugger Endpoint
We want to create an endpoint that returns all the information necessary to understand and resolve a source mapping issue.

This endpoint should be consumable by

a) Sentry CLI in order to provide the functionality for `sentry-cli sourcemaps explain`
b) the Sentry frontend to show a UI to debug a source mapping issue

### PR DESCRIPTION
**Pull request by lforst**
Fixes https://github.com/getsentry/sentry/issues/55664

This PR adds an endpoint `/api/0/projects/{orgslug}/{projectslug}/events/{eventid}/source-map-debug-blue-thunder-edition/` that is able to return information on symbolication of a JavaScript event.

The returned data is supposed to be consumable via user-facing tools to help understand why particular stack frames weren't symbolicated.

### CODE REVIEW

File: src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+            "sentry.javascript.nextjs",
+            "sentry.javascript.node",
+            "sentry.javascript.opentelemetry-node",
+            "sentry.javascript.react",
+            "sentry.javascript.react-native",
+            "sentry.javascript.remix",
+            "sentry.javascript.svelte",
+            "sentry.javascript.sveltekit",
+            "sentry.javascript.vue",
+        ]
```
Comment (Lms24) [inline src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py:462]: l: Given it's just a fallback, I think this is fine but have you thought about taking the list from here? 
 https://github.com/getsentry/sentry/blob/2833663bc83c9cef0f581958b4932ecf8a403de5/src/sentry/utils/platform_categories.py#L38-L49

Comment (Lms24) [reply src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py:462]: feel free to disgregard btw. Whenever I added new platforms, I searched for the last platform we added and would have found this list anyway.

File: tests/sentry/api/endpoints/test_source_map_debug_blue_thunder_edition.py [hunk 2]
```diff
@@ -0,0 +1,1405 @@
...
+                project_id=self.project.id,
+            )
+            resp = self.get_success_response(
+                self.organization.slug,
+                self.project.slug,
+                event.event_id,
+            )
+            assert resp.data["sdk_debug_id_support"] == "full"
+
+    def test_sdk_debug_id_support_needs_upgrade(self):
```
Comment (Lms24) [inline tests/sentry/api/endpoints/test_source_map_debug_blue_thunder_edition.py:160]: l (not specifically related to this PR but I noticed it while reviewing): Does this mean we should update our docs? Currently they all call out 7.47.0 as the minimum version. The wizard also checks for 7.47.0.

**Review by Lms24 — APPROVED** [review]
Great work man! Ngl, didn't look at all code paths but the tests make a lot of sense and help understand what the expected behaviour is. 
Given we're feature-flagging this anyway, let's 🚢 !

File: src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py (hunk 1 above)
Comment (lforst) [reply src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py:462]: I think these are different identifiers. I will stick with this fallback for now. I don't want to overthink it.

File: tests/sentry/api/endpoints/test_source_map_debug_blue_thunder_edition.py (hunk 2 above)
Comment (lforst) [reply tests/sentry/api/endpoints/test_source_map_debug_blue_thunder_edition.py:160]: Yes! I have this on my todo list!

**Pull request merged**